Source/Core/Core/HW/BBA-TAP/TAP_Win32.cpp

```diff
@@ -242,19 +263,19 @@ bool CEXIETHERNET::IsActivated()
 
 bool CEXIETHERNET::SendFrame(u8 *frame, u32 size)
 {
-	DEBUG_LOG(SP1, "SendFrame %x\n%s",
-		size, ArrayToString(frame, size, 0x10).c_str());
+	DEBUG_LOG(SP1, "SendFrame %u\n%s", size, ArrayToString(frame, size, 0x10).c_str());
 
-	OVERLAPPED overlap;
-	ZeroMemory(&overlap, sizeof(overlap));
+	// Make sure our first write is done before we start the second
+	WaitForSingleObject(mHWriteEvent, INFINITE);
 
 	// WriteFile will always return false because the TAP handle is async
-	WriteFile(mHAdapter, frame, size, NULL, &overlap);
-
-	DWORD res = GetLastError();
-	if (res != ERROR_IO_PENDING)
+	if (!WriteFile(mHAdapter, frame, size, NULL, &mWriteOverlapped))
```

Comment:
From the [docs on WriteFile](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365747%28v=vs.85%29.aspx):
> Accessing the output buffer while a write operation is using the buffer may lead to corruption of the data written from that buffer. Applications must not write to, reallocate, or free the output buffer that a write operation is using until the write operation completes.
I'm not sure about the semantics of the caller, but is seems like this code doesn't do anything to guarantee this?